storage/reads/reader_test.go

```diff
@@ -0,0 +1,124 @@
+package reads_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/execute/executetest"
+	"github.com/influxdata/flux/memory"
+	"github.com/influxdata/influxdb/flux/stdlib/influxdata/influxdb"
+	"github.com/influxdata/influxdb/mock"
+	"github.com/influxdata/influxdb/storage/reads"
+	"github.com/influxdata/influxdb/storage/reads/datatypes"
+	"github.com/influxdata/influxdb/tsdb/cursors"
+)
+
+func TestDuplicateKeys_ReadFilter(t *testing.T) {
+	closed := 0
+
+	s := mock.NewStoreReader()
+	s.ReadFilterFunc = func(ctx context.Context, req *datatypes.ReadFilterRequest) (reads.ResultSet, error) {
+		inputs := make([]cursors.Cursor, 2)
+		inputs[0] = func() cursors.Cursor {
+			called := false
+			cur := mock.NewFloatArrayCursor()
+			cur.NextFunc = func() *cursors.FloatArray {
+				if called {
+					return &cursors.FloatArray{}
+				}
+				called = true
+				return &cursors.FloatArray{
+					Timestamps: []int64{0},
+					Values:     []float64{1.0},
+				}
+			}
+			cur.CloseFunc = func() {
+				closed++
+			}
+			return cur
+		}()
+		inputs[1] = func() cursors.Cursor {
+			called := false
+			cur := mock.NewIntegerArrayCursor()
+			cur.NextFunc = func() *cursors.IntegerArray {
+				if called {
+					return &cursors.IntegerArray{}
+				}
+				called = true
+				return &cursors.IntegerArray{
+					Timestamps: []int64{10},
+					Values:     []int64{1},
+				}
+			}
+			cur.CloseFunc = func() {
+				closed++
+			}
+			return cur
+		}()
```

Comment:
Do those 2 tables have the same group key?

That's why you expect to get only the first one?

Reply:
Ah yes. That must be what this is. I wasn't sure exactly what was going on here.